x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/HttpCertificateCommand.java

```diff
@@ -0,0 +1,1149 @@
+        terminal.println("These hostnames will be added as \"DNS\" names in the \"Subject Alternative Name\"");
+        terminal.println("(SAN) field in your certificate.");
+        terminal.println("");
+        terminal.println("You should list every hostname and variant that people will use to connect to");
+        terminal.println("your cluster over http.");
+        terminal.println("Do not list IP addresses here, you will be asked to enter them later.");
+        terminal.println("");
+        terminal.println("If you wish to use a wildcard certificate (for example *.es.example.com) you");
+        terminal.println("can enter that here.");
+        terminal.println("");
+        terminal.println("When you have entered all the hostnames that you need, press <ENTER> to move on");
+        terminal.println("to the next step.");
+        terminal.println("");
+
+        final List<String> dnsNames = new ArrayList<>();
+        while (true) {
+            dnsNames.addAll(readMultiLineInput(terminal));
+            if (dnsNames.isEmpty()) {
+                terminal.println("You did not enter any hostnames.");
+                terminal.println("Clients are likely to encounter TLS hostname verification errors if they");
+                terminal.println("connect to your cluster using a DNS name.");
+            } else {
+                terminal.println("You entered the following hostnames.");
+                terminal.println("");
+                dnsNames.forEach(s -> terminal.println(" - " + s));
+            }
+            terminal.println("");
+            if (terminal.promptYesNo("Is this correct", true)) {
+                break;
+            } else {
+                dnsNames.clear();
+            }
+        }
+
+        printHeader("Which IP addresses will be used to connect to " + nodeDescription + "?", terminal);
+        terminal.println("If your clients will ever connect to your nodes by numeric IP address, then you");
+        terminal.println("can list these as valid IP \"Subject Alternative Name\" (SAN) fields in your");
+        terminal.println("certificate.");
+        terminal.println("");
+        terminal.println("If you do not have fixed IP addresses, or not wish to support direct IP access");
+        terminal.println("to your cluster then you can just press <ENTER> to skip this step.");
+        terminal.println("");
+
+        List<String> ipNames;
+        while (true) {
+            ipNames = readMultiLineInput(terminal);
+            if (ipNames.isEmpty()) {
+                terminal.println("You did not enter any IP addresses.");
+            } else {
+                terminal.println("You entered the following IP addresses.");
+                terminal.println("");
+                ipNames.forEach(s -> terminal.println(" - " + s));
+            }
+            terminal.println("");
+            if (terminal.promptYesNo("Is this correct", true)) {
+                break;
+            }
```

Comment:
This wasn't actually needed because dnsNames and ipNames were handled differently (final, allAll etc).
But I've removed that distinction since it was confusing.